mtags/src/main/scala/scala/meta/internal/metals/Docstrings.scala

```diff
@@ -39,37 +39,26 @@ class Docstrings(index: GlobalSymbolIndex) {
     cache.get(symbol) match {
       case Some(value) =>
         if (value == EmptySymbolDocumentation) Optional.empty()
-        else Optional.of(value)
+        else if (value.docstring.nonEmpty) Optional.of(value)
+        else
+          value match {
+            case value: MetalsSymbolDocumentation =>
+              Optional.of(parentDocumentation(symbol, value, parents))
+            case _ => Optional.of(value)
+          }
       case None =>
         indexSymbol(symbol)
         val result = cache.get(symbol)
         val resultWithDocs = result match {
           case None =>
             cache(symbol) = EmptySymbolDocumentation
             result
-          /* Fall back to parent scaladocs if nothing is specified for the current symbol
+          /* Fall back to parent javadocs/scaladocs if nothing is specified for the current symbol
            * This way we also cache the result in order not to calculate parents again.
            */
           case Some(value: MetalsSymbolDocumentation)
               if value.docstring.isEmpty() =>
-            parents
-              .parents()
-              .asScala
-              .flatMap { s =>
-                if (cache.contains(s)) cache.get(s)
-                else {
-                  indexSymbol(s)
-                  cache.get(s)
-                }
-              }
-              .find(_.docstring().nonEmpty)
-              .fold {
-                result
-              } { withDocs =>
-                val updated = value.copy(docstring = withDocs.docstring())
-                cache(symbol) = updated
-                Some(updated)
-              }
+            Some(parentDocumentation(symbol, value, parents))
```

Comment:
This looks like retry of `documentation` after index, so maybe we can add a flag and call the  `documentation` function again?